optimization debian package manager tweaks

rajpratik71 opened the pull request:
By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

ianpartridge commented on 5.0/ubuntu/16.04/slim/Dockerfile:
```diff
@@ -3,7 +3,7 @@ LABEL maintainer="Swift Infrastructure <swift-infrastructure@swift.org>"
 
 RUN export DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true && apt-get -q update && \
-    apt-get -q install -y \
+    apt-get --no-install-recommends install -y apt-utils ca-certificates \
```
Please put the `apt-utils` and `ca-certificates` on their own line, to match the rest of the packages.

ianpartridge reviewed (commented):
In principle, great idea!

How many unnecessary packages does this remove, and what effect does it have on final image size?

rajpratik71 committed:
optimization debian package manager tweaks

By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Also , added packages apt-utils ca-certificates

Because build is

1.  Slow because "apt-utils" not installed

2. to avoid build to exits with error without having certificate in wget , curl or even git clone

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

albertoblue87-netizen reviewed (commented):
Done
